micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/ServiceRestClientConfigurationSupport.groovy

```diff
@@ -120,8 +122,17 @@ class ServiceRestClientConfigurationSupport {
     }
 
     @Bean
-    MetricsAspect metricsAspect(MetricRegistry metricRegistry) {
-        return new MetricsAspect(metricRegistry)
+    @ConditionalOnExpression('${rest.client.metrics.enabled:true}')
+    RestOperationsMetricsAspect restOperationsMetricsAspect(
+            MetricRegistry metricRegistry, URIMetricNamer uriMetricNamer
+    ) {
+        return new RestOperationsMetricsAspect(metricRegistry, uriMetricNamer)
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(URIMetricNamer)
```

Comment:
This AFAIR is supported (guaranteed) only when used in the AutoConfiguration mechanism. But if it worked for you (you were able to override it) then we will be worried about on the first (potential) production failure.